iis-wf/iis-wf-import/src/test/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceJobTest.java

```diff
@@ -0,0 +1,187 @@
+    
+    private static final String OUTPUT_NAME_DOCMETA = "docmeta";
+    private static final String OUTPUT_NAME_DATASET = "dataset";
+    private static final String OUTPUT_NAME_PROJECT = "project";
+    private static final String OUTPUT_NAME_ORGANIZATION = "organzation";
+    private static final String OUTPUT_NAME_DOC_PROJ = "doc-proj";
+    private static final String OUTPUT_NAME_PROJ_ORG = "proj-org";
+    private static final String OUTPUT_NAME_DEDUP = "dedup";
+    
+
+    private ClassLoader cl = getClass().getClassLoader();
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        SparkConf conf = new SparkConf();
+        conf.setAppName(ImportInformationSpaceJobTest.class.getSimpleName());
+        conf.setMaster("local");
+        conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
+        conf.registerKryoClasses(OafModelUtils.provideOafClasses());
+
+        spark = SparkSession.builder().config(conf).getOrCreate();
+
+    }
+
+    @Before
+    public void before() throws IOException {
+        workingDir = Files.createTempDirectory("test_import_info_space");
+        inputDir = workingDir.resolve("input");
+        inputGraphDir = inputDir.resolve("graph");
+
+        outputDir = workingDir.resolve("output");
+        outputReportDir = workingDir.resolve("output_report");
+    }
+
+    @After
+    public void after() throws IOException {
+        FileUtils.deleteDirectory(workingDir.toFile());
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        spark.stop();
+    }
+    
+    @Test
+    public void testImportFromTextGraph() throws Exception {
+        testImportFromGraph(false);
+    }
+    
+    @Test
+    public void testImportFromParquetGraph() throws Exception {
+        testImportFromGraph(true);
+    }
+    
+    private void testImportFromGraph(boolean isParquetFormat) throws Exception {
+        
+        @SuppressWarnings("unchecked")
+        Class<? extends Oaf> graphClasses[] = new Class[] {
+                eu.dnetlib.dhp.schema.oaf.Dataset.class,
+                eu.dnetlib.dhp.schema.oaf.Organization.class,
+                eu.dnetlib.dhp.schema.oaf.OtherResearchProduct.class,
+                eu.dnetlib.dhp.schema.oaf.Software.class,
+                eu.dnetlib.dhp.schema.oaf.Publication.class,
+                eu.dnetlib.dhp.schema.oaf.Project.class,
+                eu.dnetlib.dhp.schema.oaf.Relation.class
+        };
+        
+        for (Class<? extends Oaf> graphClass : graphClasses) {
+            String graphTableName = graphClass.getSimpleName().toLowerCase();
+            String inputGraphTableJsonDumpPath = String.format("%s/%s.json",
+                    "eu/dnetlib/iis/wf/importer/infospace/input/graph", graphTableName);
+            createGraphTableFor(inputGraphTableJsonDumpPath, graphTableName, graphClass, isParquetFormat);
+        }
+        
+        // when
+        ImportInformationSpaceJob.main(new String[]{
```

Comment:
General note: this way of testing spark jobs makes does not use `spark_utils` and we could think of slowly removing `spark_utils` from tests.

Reply:
I'd probably stick to the way we test spark jobs within the IIS if there was no need to rely on `SparkSession` what made using `SparkJobExecutor` and the stack behind it pretty cumbersome.